analysis/task_analysis_volume/group_level_analysis_vol.ipynb

```diff
@@ -0,0 +1,1013 @@
+    "    \n",
+    "    df_subs = pd.DataFrame(subs,columns=['subs'])\n",
+    "    non_img_data = pd.read_csv(f\"../../../sourcedata/non_imaging_data/MMJ-Processed_data-2022_05_27-13_58-6858bbe.csv\",low_memory=False)\n",
+    "    dummy_dfs = []\n",
+    "    \n",
+    "    if 'sex' not in covariates:\n",
+    "        #create design matrix with 1 column of 1s and as many rows as there are subjects\n",
+    "        group_design_matrix = pd.DataFrame([1] * len(subs), columns=['group_average'],)\n",
+    "    \n",
+    "    else:\n",
+    "        #create empty design matrix to add to\n",
+    "        group_design_matrix = pd.DataFrame()\n",
+    "        #add columns for male and female, that will then be combined to create the group average \n",
+    "        grouped_sex = non_img_data.groupby(\"IDS.CHR.Subject\")[\"SBJ.CHR.Sex\"].agg(\"first\")\n",
+    "        dict_sex = grouped_sex.to_dict()\n",
+    "        df_subs['sex'] = df_subs['subs'].map(dict_sex)\n",
+    "        dummy_dfs.append(pd.get_dummies(df_subs['sex']))\n",
+    "    \n",
+    "    #numericals can be added directly\n",
+    "    #mean center numerical values like age and CUDIT!\n",
+    "\n",
+    "    if 'age' in covariates:\n",
+    "        #numerical can be added directly\n",
+    "        #mean center numerical value\n",
+    "        grouped_age = non_img_data.groupby(\"IDS.CHR.Subject\")[\"SBJ.INT.Age\"].agg(\"first\")\n",
+    "        dict_age = grouped_age.to_dict()\n",
+    "        group_design_matrix['age'] = df_subs['subs'].map(dict_age)\n",
+    "        group_design_matrix['age'] = group_design_matrix['age'] - group_design_matrix['age'].mean()\n",
+    "    \n",
+    "    if 'total_cudit' in covariates:\n",
+    "        if group == 'HC': \n",
+    "            grouped_HC_baseline_cudit = non_img_data[non_img_data['SSS.CHR.Time_point'] == 'Screening'].groupby('IDS.CHR.Subject')['INV.INT.CUDIT.Summed_score'].agg(\"first\")\n",
+    "            dict_HC_baseline_cudit = grouped_HC_baseline_cudit.to_dict()\n",
+    "            group_design_matrix['total_cudit'] = df_subs['subs'].map(dict_HC_baseline_cudit)\n",
+    "            group_design_matrix['total_cudit'] = group_design_matrix['total_cudit'] - group_design_matrix['total_cudit'].mean()\n",
+    "\n",
+    "        else:\n",
+    "            if ses == 'baseline':\n",
+    "                dict_MM_baseline_cudit = non_img_data[non_img_data['SSS.CHR.Time_point'] == 'Baseline'].groupby('IDS.CHR.Subject')['INV.INT.CUDIT.Summed_score'].agg(\"first\").to_dict()\n",
+    "                group_design_matrix['total_cudit'] = df_subs['subs'].map(dict_MM_baseline_cudit)\n",
+    "                group_design_matrix['total_cudit'] = group_design_matrix['total_cudit'] - group_design_matrix['total_cudit'].mean()\n",
+    "                #needed because MM_141 is nan, so replacing with 0, which is the mean\n",
+    "                group_design_matrix['total_cudit'].fillna(0, inplace=True)\n",
```

Comment:
missing issue. check.

Reply:
I was doing imputation by the mean -- it's not the best but it's not a major part of the analysis and i thought it would be ok. what do you suggest instead?